<launch>
  <arg name="session_mode" default="runtime" />
  <arg name="spawn_x" default="0.0" />
  <arg name="spawn_y" default="0.0" />
  <arg name="spawn_z" default="0.2" />
  <arg name="robot_name" default="smart" />
  <arg name="config_file" default="$(find simulation_bringup)/config/smart_control_config.yaml" />
  <arg name="world_file" default="$(find gazebo_ros)/worlds/empty.world" />
  <arg name="urdf_file" default="$(find simulation_bringup)/models/stage07_controlled_car/urdf/smart_controlled.xacro" />

  <include file="$(find simulation_bringup)/launch/stage07_controlled_car.launch">
    <arg name="session_mode" value="$(arg session_mode)" />
    <arg name="spawn_x" value="$(arg spawn_x)" />
    <arg name="spawn_y" value="$(arg spawn_y)" />
    <arg name="spawn_z" value="$(arg spawn_z)" />
    <arg name="robot_name" value="$(arg robot_name)" />
    <arg name="config_file" value="$(arg config_file)" />
    <arg name="world_file" value="$(arg world_file)" />
    <arg name="urdf_file" value="$(arg urdf_file)" />
  </include>

  <node pkg="simulation_bringup" type="cmdvel2gazebo.py" name="cmdvel2gazebo" output="screen">
    <param name="robot_name" value="$(arg robot_name)" />
    <param name="cmd_topic" value="/$(arg robot_name)/cmd_vel" />
    <param name="controller_namespace" value="/$(arg robot_name)" />
  </node>
</launch>
